azure-pipelines/release.yml

```diff
@@ -15,66 +15,100 @@ parameters:
+          Write-Host "Publish to pre-release channel."
+        } ElseIf ("$(resources.pipeline.officialBuildCI.sourceBranch)" -eq "refs/heads/release") {
+          Write-Host "Publish to release channel."
+        } Else {
+          throw "Unexpected branch name: $(resources.pipeline.officialBuildCI.sourceBranch)."
+        }
+        $additionalPublishArgs += '--packagePath'
 
-      $platforms = "arm64", "x64", "ia32"
-      $allVsixs = Get-ChildItem $publishArtifacts *.vsix
-      foreach ($vsix in $allVsixs) {
-        foreach ($plat in $platforms) {
-          if ($vsix.Name.Contains($plat)) {
-              $additionalPublishArgs += $vsix
+        $platforms = "arm64", "x64", "ia32"
+        $allVsixs = Get-ChildItem $publishArtifacts *.vsix
+        foreach ($vsix in $allVsixs) {
+          foreach ($plat in $platforms) {
+            if ($vsix.Name.Contains($plat)) {
+                $additionalPublishArgs += $vsix
+            }
           }
         }
-      }
 
-      Write-Host "Command run is: vsce $($additionalPublishArgs)."
-      If ("${{ parameters.test }}" -eq "true") {
-        Write-Host "In test mode, command is printed instead of run."
-        Write-Host "🔒 Verify PAT."
-        vsce verify-pat ms-dotnettools
-      }
-      Else {
-        vsce @additionalPublishArgs
-      }
-    displayName: 🚀 Publish to Marketplace
-    workingDirectory: $(Pipeline.Workspace)
-    env:
-      VSCE_PAT: $(VSCodeMarketplacePAT)
+        Write-Host "Command run is: vsce $($additionalPublishArgs)."
+        If ("${{ parameters.test }}") {
+          Write-Host "In test mode, command is printed instead of run."
+          Write-Host "🔒 Verify PAT."
+          vsce verify-pat ms-dotnettools
+        }
+        Else {
+          vsce @additionalPublishArgs
+        }
+      displayName: 🚀 Publish to Marketplace
+      workingDirectory: $(Pipeline.Workspace)
+      env:
+        VSCE_PAT: $(VSCodeMarketplacePAT)
+- stage: 'TagReleaseNumberToRoslyn'
+  displayName: 'Tag release number of vscode-csharp to Roslyn'
+  dependsOn: 'PublishStage'
+  jobs:
+  - job: 'TagRoslyn'
+    pool:
+      vmImage: ubuntu-latest
+    steps:
+    - task: NodeTool@0
+      displayName: 'Install Node.js 18.x'
+      inputs:
+        versionSpec: '18.x'
+    - checkout: self
+      clean: true
+      submodules: true
+      fetchTags: false
+      fetchDepth: 0
+    - pwsh: |
+        # Need these two files to find the roslyn commit for the given branch
+        git checkout $(resources.pipeline.officialBuildCI.sourceCommit) -- ./package.json ./server/NuGet.config
+      displayName: 'Checkout package.json and NuGetConfig from targeting branch'
+    - pwsh: |
+        npm ci
+        npm install
+        npm install -g gulp
+        # build number is generated by nerdbank git versioning and set to AzureDev
+        gulp createTags --releaseVersion $(resources.pipeline.officialBuildCI.runName) --releaseCommit $(resources.pipeline.officialBuildCI.sourceCommit)
```

Comment:
Should we skip this (or have some kind of dry run) if this is a test run?

Reply:
I passed the `${{ paramters.test }}` into the task and skip the tagging. (Only print the commit found with the nuget packages)